src/components/schedule-page/table/schedule/schedule-parts/base-cell/use-component-visible.tsx

```diff
@@ -0,0 +1,26 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { useState, useEffect, useRef } from "react";
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+export default function useComponentVisible(initialIsVisible: boolean): any {
+  const [isComponentVisible, setIsComponentVisible] = useState(initialIsVisible);
+  const ref = useRef(null);
+
+  const handleClickOutside = (event): void => {
+    if (ref.current && !(ref.current! as any).contains(event.target)) {
+      setIsComponentVisible(false);
+    }
+  };
+
+  useEffect((): any => {
+    document.addEventListener("click", handleClickOutside, true);
+    return (): any => {
+      document.removeEventListener("click", handleClickOutside, true);
+    };
+  });
```

Comment:
tutaj pewnie react narzeka, że nie ma dependency array, a kod w use effect zależy od handleClickOutside.
Jeśli byś poprostu to dodała, to by się za często rerenderował (infinite loop), więc wyjścia są dwa:
* albo wciągnąć handleClickOutside do środka useEffect - to chyba bym wybrał (więcej o useEffect: https://reactjs.org/docs/hooks-reference.html#useeffect)
* albo funkcję handleClickOutside otoczyć hookiem useCallback - który spowoduje, że handleClickOutside nie będzie się tworzył nowy przy każdym rerenderze  (https://reactjs.org/docs/hooks-reference.html#usecallback)

Reply:
wrzuciłam handleClickOutside do środka useEffect